sr_robot_lib/include/sr_robot_lib/MST.hpp

```diff
@@ -11,67 +11,101 @@
+    * @param nh ROS node handle.
+    * @param device_id The device id.
+    * @param update_configs_vector A vector containing information regarding which type of data to poll.
+    * @param update_state Defines current operation state of the sensor (INITIALIZATION or OPERATION).
+    * @param init_tactiles_vector A generic tactile data vector containing sensor information (e.g. Serial Number).
+    */
   MST(ros::NodeHandle nh, std::string device_id, std::vector<generic_updater::UpdateConfig> update_configs_vector,
        operation_mode::device_update_state::DeviceUpdateState update_state,
        boost::shared_ptr<std::vector<GenericTactileData> > init_tactiles_vector);
 
-  /*
-   * This function is called each time a new etherCAT message
-   * is received in the sr08.cpp driver. It  updates the tactile
-   * sensors values contained in sensor_data.
-   *
-   * @param status_data the received etherCAT message
-   */
+  /**
+    * Initialise relevant MST sensor data arrays.
+    */
+  void initialise_tactile_data_structure(boost::shared_ptr<std::vector<GenericTactileData> > init_tactiles_vector);
+
+  /**
+    * Extract the tactile specific data from the incoming EtherCAT status message from the Palm.
+    * And update the tactile data structure.
+    *
+    * @param status_data Is a pointer to an incoming EtherCAT status message from the Palm.
+    */
   virtual void update(StatusType *status_data);
 
-  /*
-   * Publish the information to a ROS topic.
-   *
-   */
+  /**
+    * Publish MST tactile data. 
+    * Not implemented, as this is published from the controller publisher.
+    * But still necessary to keep it in the class, as it maybe be called by the hand drivers
+    */
   virtual void publish();
 
-  /*
-   * This function adds the diagnostics for the tactiles to the
-   * multi diagnostic status published by the hand.
-   */
-  virtual void add_diagnostics(std::vector<diagnostic_msgs::DiagnosticStatus> &vec,
-                               diagnostic_updater::DiagnosticStatusWrapper &d);
+  /**
+    *  This function appends MST specific diagnostic data to the runtime_monitor node of a running Hand.
+    *
+    * @param diagnostic_vector The vector of diagnostic messages which to append the sensor's diagnostic data.
+    * @param diagnostic_status_wrapper A diagnostic status wrapper used to update the sensor's diagnostics with.
+    */
+  virtual void add_diagnostics(std::vector<diagnostic_msgs::DiagnosticStatus> &diagnostic_vector,
+                               diagnostic_updater::DiagnosticStatusWrapper &diagnostic_status_wrapper);
+
+  /**
+    * Getter for the current MST tactile data
+    * 
+    * @return A vector of tactiles::AllTactileData messages, updated with MST tactile data
+    */
+  virtual std::vector<AllTactileData> *get_tactile_data();
 
 private:
-  sr_robot_msgs::MSTPalm sensor_data;
-  std::shared_ptr<ros::Publisher> publisher;
+  /// Stores current MST tactile data
+  sr_robot_msgs::MSTAll sensor_data;
```

Comment:
This is a member variable. It should have a trailing underscore.

Reply:
Updated here: https://github.com/shadow-robot/sr-ros-interface-ethercat/pull/265/commits/fb95ab9c31ed480ba65f9b69c9c47b3f8b5034ca